src/modules/commander/state_machine_helper.cpp

```diff
@@ -705,7 +734,17 @@ bool check_invalid_pos_nav_state(vehicle_status_s &status, bool old_failsafe, or
 	}
 
 	if (fallback_required) {
-		if (use_rc) {
+		// If GNSS-denied, the estimator_selector will attempt to enable GNSS in case of a failsafe
+		// We do not want to terminate until we are certain that enabling GNSS did not help
+		if (hrt_elapsed_time(&last_gnss_denied_ts) < 1_s) {
+			PX4_WARN("Invalid pos_nav_state but GNSS denied");
```

Comment:
This log line should be removed to avoid spam?

Reply:
Good catch. I used it for debugging and forgot to remove it